### PR TITLE
Use correct time separator for Indonesian locale

### LINKED ISSUE #104069: Time picker with incorrect time separator format for locale `id`
Should be `.`, not `:`.

b/229057815
b/231987861

### PR DESCRIPTION
**Pull request by guidezpl**
Fixes #104069

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_localizations/lib/src/l10n/README.md [hunk 1]
```diff
@@ -159,30 +159,29 @@ section in the Material spec. The Material theme uses the
...
 You can see what that script would generate by running this command:
 
 ```dart
-dart dev/tools/localizations/bin/gen_localizations.dart packages/flutter_localizations/lib/src/l10n material
+dart dev/tools/localization/bin/gen_localizations.dart packages/flutter_localizations/lib/src/l10n material
 ```
 
-The gen_localizations app just combines the contents of all of the
+The gen_localizations script just combines the contents of all of the
 .arb files into a single `Map` per library that has entries for each .arb
```
Comment (HansMuller) [inline packages/flutter_localizations/lib/src/l10n/README.md:171]: Really still a Map?

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter_localizations/lib/src/l10n/README.md (hunk 1 above)
Comment (guidezpl) [reply packages/flutter_localizations/lib/src/l10n/README.md:171]: Not anymore, fixed

File: packages/flutter_localizations/lib/src/l10n/README.md [hunk 2]
```diff
@@ -155,36 +152,32 @@ section in the Material spec. The Material theme uses the
...
+2. The file name prefix (the file name less the locale
 suffix) for the .arb files in this directory.
 
-To in-place update the generated localizations file using the default
-values, you can just run:
-
-```dart
-dart dev/tools/localizations/bin/gen_localizations.dart --overwrite
-```
-
```
Comment (Hixie) [inline packages/flutter_localizations/lib/src/l10n/README.md:181]: thank you so much for updating the documentation

**Pull request merged**